Fixes #1514. Enable latest SNAPSHOT option by default

eed3si9n opened the pull request:
Fixes #1514
Fixes #1616
#1520 originally fixed #1514 by reimplementing part of chain resolver to check all resolvers to find the latest snapshot artifacts. This behavior did not work well:
1. with Maven repositories where sbt was failing to calculate the correct publication dates.
2. with module doesn't have any artifacts (parent POMs). #1616 
#2075 should fix the publication date issue, and  we should enable this flag back again. 1bf789eb3dc48f8786da326f2ce18793f09faea5 fixes the module without artifacts.

The build user can opt out by:

``` scala
updateOptions := updateOptions.value.withLatestSnapshots(false)
```

/review @jsuereth, @dwijnand, @Duhemm

eed3si9n committed:
Fixes #1514. Enable latest SNAPSHOT option by default

#1520 originally fixed #1514 by reimplementing part of chain resolver
to check all resolvers to find the latest snapshot artifacts.
This behavior did not work well with Maven repositories where sbt was
failing to calculate the correct publication dates.
Now that #2075 fixes the Maven integration issue we should enable this
flag back again.

The build user can opt out by:

    updateOptions := updateOptions.value.withLatestSnapshots(false)

dwijnand commented on sbt/src/sbt-test/dependency-management/snapshot-resolution/build.sbt:
```diff
@@ -37,9 +37,9 @@ lazy val dependent = project.
+    libraryDependencies += "com.badexample" % "badexample" % "1.0-SNAPSHOT"
     // Setting this to false fails the test
-    updateOptions := updateOptions.value.withLatestSnapshots(true)
+    // updateOptions := updateOptions.value.withLatestSnapshots(true)
```
why the commented out code?

eed3si9n replied:
This option has gone back and forth, so I am leaving that in in case we're going to default it to `false` again.

dwijnand replied:
alright

dwijnand commented:
LGTM asides from the minor commented code comment

eed3si9n committed:
Bump up ivy to 2.3.0-sbt-39ece6f17e1af62c2e89a9d88b59c0c3aad56145

Ref sbt/ivy#21

eed3si9n committed:
Fixes #1616. Fixes the "latest snapshot" chain resolver

The "latest snapshot" chain resolver was assuming that there's at least
one artifact per module. I think that was the root cause of #1616.

eed3si9n committed:
Bump up ivy to 2.3.0-sbt-2cc8d2761242b072cedb0a04cb39435c4fa24f9a
